### DISCUSSION TITLE
cloud s3 Class A operator

### DISCUSSION THREAD

**Discussion by big-thousand** [milvus-io/milvus#20423] [Q&A and General discussion]
Most of cloud s3 provider will billing for Class A operator.
It seems that milvus will always perform Class A operations even when no data and no client
<img width="1213" alt="image" src="https://user-images.githubusercontent.com/114633619/200729992-79270e32-f532-43cf-9954-81489ca84b4a.png">
Whether I can config something to avoid this ?

**Comment by LoveEachDay** [comment]
@big-thousand Could you provide more information about operations class? Seems like this classification is about s3 api call.

**Reply by big-thousand** [reply]
I use milvus on k8s
```
# milvus operator
helm repo add milvus-operator https://milvus-io.github.io/milvus-operator/
helm repo add milvus https://milvus-io.github.io/milvus-helm/
helm repo update
helm upgrade milvus-operator milvus-operator/milvus-operator --install -n milvus-operator --create-namespace     
# milvus cluster
RELEASE=milvus-cluster MILVUS_CLUSTER_NAMESPACE=milvus-cluster-default
helm upgrade ${RELEASE} milvus/milvus --install -n ${MILVUS_CLUSTER_NAMESPACE} --create-namespace \
    --set minio.enabled=false \
    --set externalS3.enabled=true \
    --set externalS3.host="xxxxxx.r2.cloudflarestorage.com" \
    --set externalS3.port="443" \
    --set externalS3.accessKey="xxxxxxxx" \
    --set externalS3.secretKey="xxxxxxxx" \
    --set externalS3.useSSL=true \
    --set externalS3.bucketName="milvus-test"
```

**Reply by big-thousand** [reply]
Class A operator is in cloudflare r2 doc
https://developers.cloudflare.com/r2/platform/pricing/
https://cloud.google.com/storage/pricing

**Reply by big-thousand** [reply]
@LoveEachDay any update?

**Reply by LoveEachDay** [reply]
@big-thousand Milvus have some background tasks which will take some class A operations:
1> compaction
compaction will combine small segments into a bigger one (you can search fast with big segment). In compaction, milvus will first read back the segment contents from object store, and save a new segment which will store back to object store. Then, milvus will read the new segment back into the indexnode to build index, which will store the index files back to the object store. 

2> gc
compaction will generate a bigger segments from the small one, `gc` will then delete the old smaller segments. And `gc` will also helps to delete segments when you drop a collection.

After those two operations are down, it won't have any operations for your object store.
